rust/src/container.rs

```diff
@@ -387,6 +398,31 @@ pub fn container_encapsulate(args: Vec<String>) -> CxxResult<()> {
         })
         .collect::<Result<_>>()?;
 
+    let package_structure = match &opt.ocimeta_json {
+        Some(ocimeta_json) => {
+            let mut annotation_build: Vec<Vec<String>> = Vec::new();
+            let path = std::path::Path::new(ocimeta_json);
+            if path.is_file() {
+                let manifest = oci_spec::image::ImageManifest::from_file(path).unwrap();
+                let layers = manifest.layers();
+                for layer in layers {
+                    let annotation_layer = layer
+                        .annotations()
+                        .as_ref()
+                        .expect("Layer does not consist annotation");
+                    let pkgs: Vec<&str> = annotation_layer["Content"].split(',').collect();
```

Comment:
I think we should define `Content` as a const in ostree-ext at least, since it's used in both places.  But more generally, it seems a lot simpler to just pass the prior manifest into ostree-ext.  Then the reading and writing is in the same place.

IOW starting with

```
diff --git a/lib/src/container/encapsulate.rs b/lib/src/container/encapsulate.rs
index 3418b6e2..58730ea3 100644
--- a/lib/src/container/encapsulate.rs
+++ b/lib/src/container/encapsulate.rs
@@ -391,7 +391,7 @@ pub struct ExportOpts {
     /// Use only the standard OCI version label
     pub no_legacy_version_label: bool,
     /// Prevent major change in packaging structure by taking previous builds in order of priority
-    pub prior_build_metadata: Option<Vec<Vec<String>>>,
+    pub prior_build_metadata: Option<&oci_spec::image::ImageManifest>,
 }
 
 impl ExportOpts {
```

What do you think?